src/randomizer.py

```diff
@@ -865,6 +865,46 @@ def minus1(node: ca.Node) -> ca.BinaryOp:
 
     return True
 
+def perm_add_mask(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> bool:
+    """Add a mask of 0xFF[FFFFFFFFFFFFFF] to a random expression of integer type.
+    In some cases this mask is optimized out but affects regalloc."""
+    typemap = build_typemap(ast)
+    cands: List[Expression] = []
+
+    # Find expression to add the mask to
+    def rec(block: Block) -> None:
+        for stmt in ast_util.get_block_stmts(block, False):
+            ast_util.for_nested_blocks(stmt, rec)
+            def visitor(expr: Expression) -> None:
+                if not region.contains_node(expr):
+                    return
+                cands.append(expr)
+            replace_subexprs(stmt, visitor)
+    rec(fn.body)
+    if not cands:
+        return False
+
+    expr = random.choice(cands)
+    type: SimpleType = decayed_expr_type(expr, typemap)
+    base_type = resolve_typedefs(type, typemap)
+
+    if not isinstance(base_type, ca.TypeDecl):
+        return False
+    if not isinstance(base_type.type, ca.IdentifierType):
+        return False
+    if all(x not in base_type.type.names for x in ['int', 'char', 'long', 'short', 'unsigned']):
+        return False
+
+    # Mask as if restricting the value to 8, 16, 32, or 64-bit width.
+    # Sometimes use an unsigned mask like '0xFFu'
+    masks: List[str] = ['0xFF', '0xFFFF', '0xFFFFFFFF', '0xFFFFFFFFFFFFFFFF']
```

Comment:
`& 0xFFFFFFFFFFFFFFFF` looks ridiculous, but let's leave it in and see if it shows up in practice I guess. Did you look at whether it affected only regalloc?